src/de.rs

```diff
@@ -35,14 +35,17 @@ impl<'de> de::MapAccess<'de> for MapAccess {
     fn next_key_seed<K: de::DeserializeSeed<'de>>(&mut self, seed: K) -> Result<Option<K::Value>> {
         debug_assert!(self.next_value.is_none());
 
-        Ok(match self.iter.next().transpose()? {
-            Some(pair) => {
-                let (key, value) = convert_pair(pair);
+        // Loop until a defined value is found
```

Comment:
I don't think this is correct. If there is a Map entry that is explicitly `undefined`, it's better to pass it in to Serde IMO so that it has a chance to be handled. For example, this will break `field: Option<SomeType>` unless every such `field` has an explicit `serde(default)` annotation.